receivers/opsgenie/config.go

```diff
@@ -74,6 +86,33 @@ func NewConfig(jsonData json.RawMessage, decryptFn receivers.DecryptFunc) (Confi
 		raw.OverridePriority = &overridePriority
 	}
 
+	for idx, r := range raw.Responders {
+		if r.ID == "" && r.Username == "" && r.Name == "" {
```

Comment:
Reading the section for the `visibleTo` field and the examples for both `responders` and `visibleTo` in the [Opsgenie docs](https://docs.opsgenie.com/docs/alert-api#create-alert) it's not clear to me whether any of these three values are valid for each type.
- The `user` type goes with either `id` or `username`, but not with `name`
- The `team` type goes with either `id` or `name`, but not with `username`

This validation doesn't take these edge cases into account, could you confirm that this is not an issue?

Reply:
It looks like the `id` and `name` can be used for any types but `username` - is only for the `user` type. 

I think it makes sense to add a validation that would make sure that if username is specified then the type must be `user`. What worries me is that it would cause even bigger drift between alertmanager flavors. If you think this is ok, I will introduce it.

Reply:
If the OpsGenie API returns an error, then adding validation at the Alertmanager level is not really a breaking change, but just catching an error before it happens. I'd say double check and go for it.

Reply:
It turns out that invalid combinations of the fields are accepted by Opsgenie but ignored. So, I do not think it makes sense to add any additional validation that does not cause the request to fail

Reply:
Great, thanks for double-checking :)